walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,84 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Set;
 
+/** 
+Algorithm:
+    1. Pick time ranges from events where attendees overlap with meeting request.
+    2. Now find the non overlapping time ranges which have duration more than request duration.
+*/
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    
+    private long meetingDuration;
+    private ArrayList<TimeRange> overlappingRanges;
+    private Collection<String> requestedAttendees;
+    private Collection<TimeRange> requiredRanges;
```

Comment:
Rather than having a Collection, i'd prefer having a List/Set/Map **interface** mentioned in the declaration so that the datatype inferface we're dealing with is clear.

Reply:
Yes I get it, but the output of query method is supposed to be `Collection<TimeRange>` and input from `MeetingRequest` is a `Collection<String>`. So any suggestions for this situation?

Reply:
That should be fine. A list / set is also a collection and returning an implementation of an interface should not be a problem.
Avoid using "Collection" wherever possible since a collection could be a map or a set or a list all of them work very differently. Instead try to declare a more specific **interface (not class)** like "List" or "Set" with the "new ***" statement containing the actual implantation like ArrayList or a HashSet

Reply:
Ok, so I have changed the `Collection` to `List`.